eng/pipelines/docindex.yml

```diff
@@ -49,13 +49,30 @@ jobs:
       - template: /eng/common/pipelines/templates/steps/set-daily-docs-branch-name.yml
         parameters:
           DailyBranchVariableName: DailyDocsBranchName
+      # Sync docs fork repo onboarding files/folders
```

Comment:
Instead of cloning again here why not use a fork for both daily and main? That would enable us to use the same clone.

Reply:
Release pipeline updates Azure/main, but not azure-sdk/main. (We have mirror pipeline sync azure-sdk/main daily)
So azure-sdk/main might fail to git push Azure/main because of commits behind Azure/main.

Reply:
Using a fork for the changes in `main` would require an additional step to sync the changes into the central docs repo. I think we should keep the daily onboarding of production docs pulling from and pushing to the main repo. 

If we used a fork we'd need to sync changes from the main repo to the fork and then back again to the main repo.

Reply:
OK I believe I understand the issue now and I think using a fork can work for now.